File: bluetooth_low_energy_android/android/src/main/kotlin/dev/yanshouwang/bluetooth_low_energy/MyObject.kt
package dev.yanshouwang.bluetooth_low_energy

open class MyObject(instance: Any) {
    private val hashCode = instance.hashCode()

    override fun hashCode(): Int {
        return hashCode
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return hashCode() == other.hashCode()
    }
}

File: bluetooth_low_energy_platform_interface/lib/src/my_object.dart
abstract class MyObject {
  final int? _hashCode;

  MyObject({int? hashCode}) : _hashCode = hashCode;

  @override
  int get hashCode => _hashCode ?? super.hashCode;

  @override
  bool operator ==(Object other) {
    return other is MyObject && other.hashCode == hashCode;
  }
}
